src/XrmMockupShared/Plugin/PluginManager.cs

```diff
@@ -52,7 +52,7 @@ private void RegisterPlugins(IEnumerable<Type> basePluginTypes, Dictionary<strin
 
                 foreach (var type in proxyTypeAssembly.GetLoadableTypes())
                 {
-                    if (type.BaseType != basePluginType) continue;
+                    if (!basePluginType.IsAssignableFrom(type) || type.IsAbstract || type.GetConstructor(Type.EmptyTypes) == null) continue;
```

Comment:
As discussed in #62, there is a problem with using IsAssignableFrom, since we can't be certain that the base classes are abstract. Even the recommended base class from Microsoft isn't abstract.